.github/workflows/ci_macos.yaml

```diff
@@ -51,9 +51,15 @@ jobs:
           update_packager_index: false
 
       - name: Install boost
+        env:
+          boost_major_version: "1"
+          boost_minor_version: "84"
+          boost_patch_version: "0"
         run: |
-          boost_url="https://boostorg.jfrog.io/artifactory/main/release/1.83.0/source/boost_1_83_0.tar.gz"
-          boost_url2="https://sourceforge.net/projects/boost/files/boost/1.83.0/boost_1_83_0.tar.gz"
+          boost_version_1="${{env.boost_major_version}}.${{env.boost_minor_version}}.${{env.boost_patch_version}}"
+          boost_version_2="${{env.boost_major_version}}_${{env.boost_minor_version}}_${{env.boost_patch_version}}"
```

Comment:
I don't mind `boost_*_version` variables. But these two (`boost_version_*`) are a little redundant IMO. I would just inline them.

Reply:
>I would just inline them.

Do you mean like this?

```
$boost_url="https://boostorg.jfrog.io/artifactory/main/release/${{ env.boost_major_version }}.${{ env.boost_minor_version }}.${{ env.boost_patch_version }}/source/boost_${{ env.boost_major_version }}_${{ env.boost_minor_version }}_${{ env.boost_patch_version }}.tar.gz"
$boost_url2="https://sourceforge.net/projects/boost/files/boost/${{ env.boost_major_version }}.${{ env.boost_minor_version }}.${{ env.boost_patch_version }}/boost_${{ env.boost_major_version }}_${{ env.boost_minor_version }}_${{ env.boost_patch_version }}.tar.gz"
```

Reply:
yeah.